ports/satlovecake/SATAN LOVES CAKE.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Set XDG_DATA_HOME if not already set
+XDG_DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
+
+# Check for PortMaster installation in various directories
+if [ -d "/opt/system/Tools/PortMaster/" ]; then
+  controlfolder="/opt/system/Tools/PortMaster"
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+# PortMaster info
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
+GAMEDIR=/$directory/ports/satlovecake
+get_controls
```

Comment:
We usually have this order even if it's not a problem here.
```suggestion
get_controls
GAMEDIR=/$directory/ports/satlovecake
```

Reply:
I remade the .sh and uploaded it. I hope this corrects the things you requested fixed.🎩🙂